zkevm-circuits/src/keccak_circuit/keccak_padding.rs

```diff
@@ -0,0 +1,427 @@
+                let d_bit_0 = meta.query_advice(d_bits[8 * i], Rotation::cur());
+                constraints.push(("begin with 1", (s_i - s_i_1) * (d_bit_0 - 1u64.expr())));
+            }
+            let s_last = meta.query_advice(s_flags[s_flags.len() - 1], Rotation::cur());
+            let d_last = meta.query_advice(d_bits[KECCAK_RATE - 1], Rotation::cur());
+
+            constraints.push(("end with 1", s_last * (d_last - 1u64.expr())));
+
+            cb.add_constraints(constraints);
+            cb.gate(meta.query_selector(q_enable))
+        });
+
+        meta.create_gate("intermedium 0 checks", |meta| {
+            let mut cb = BaseConstraintBuilder::new(5);
+
+            let mut sum_padding_bits = 0u64.expr();
+            for i in 0..s_flags.len() {
+                let s_i = meta.query_advice(s_flags[i], Rotation::cur());
+                sum_padding_bits = d_bits[i * 8..(i + 1) * 8]
+                    .iter()
+                    .map(|b| meta.query_advice(*b, Rotation::cur()))
+                    .fold(sum_padding_bits, |sum, b| sum + s_i.clone() * b);
+            }
+
+            cb.add_constraint("sum padding bits == 2", sum_padding_bits - 2u64.expr());
+            cb.gate(meta.query_selector(q_enable))
+        });
+
+        meta.create_gate("input len check", |meta| {
+            let mut cb = BaseConstraintBuilder::new(5);
+
+            let mut constraints = vec![];
+            for i in 1..s_flags.len() {
+                let s_i = meta.query_advice(s_flags[i], Rotation::cur());
+                let len_i = meta.query_advice(d_lens[i], Rotation::cur());
+                let len_i_1 = meta.query_advice(d_lens[i - 1], Rotation::cur());
+                constraints.push(("len[i] = len[i-1] + !s_i", len_i - len_i_1 - not::expr(s_i)));
+            }
+
+            cb.add_constraints(constraints);
+            cb.gate(meta.query_selector(q_enable))
+        });
+
+        meta.create_gate("input rlc check", |meta| {
+            let mut cb = BaseConstraintBuilder::new(5);
+
+            let mut constraints = vec![];
+            for i in 1..s_flags.len() {
+                let s_i = meta.query_advice(s_flags[i], Rotation::cur());
+                let rlc_i = meta.query_advice(d_rlcs[i], Rotation::cur());
+                let rlc_i_1 = meta.query_advice(d_rlcs[i - 1], Rotation::cur());
+                let r = meta.query_advice(randomness, Rotation::cur());
+                let input_byte_i = d_bits[i * 8..(i + 1) * 8]
+                    .iter()
+                    .map(|bit| meta.query_advice(*bit, Rotation::cur()))
+                    .fold(0u64.expr(), |v, b| v * 2u64.expr() + b);
+                constraints.push((
+                    "rlc[i] = rlc[i-1]*r if s == 0 else rlc[i]",
+                    rlc_i.clone()
+                        - (rlc_i.clone() * s_i.clone()
+                            + (rlc_i_1 * r + input_byte_i) * not::expr(s_i.clone())),
+                ));
```

Comment:
`cb.require_equal(rlc_i, select::expr(s_i, rlc_i_1, rlc_i_1 * r + input_byte_i))`.

Reply:
Yes, should be. I don't remember which OP CODE you mentioned last time has the rlc, so I just do it literately, I could just re-use the same logic of it. 
BTW: Which OP CODE has rlc?